mono/eglib/glib.h

```diff
@@ -54,6 +54,122 @@
 #define G_END_DECLS
 #endif
 
+#ifdef __cplusplus
+
+#define g_cast monoeg_g_cast // in case not inlined (see eglib-remap.h)
+
+// g_cast converts void* to T*.
+// e.g. #define malloc(x) (g_cast (malloc (x)))
+// FIXME It used to do more. Rename?
+struct g_cast
+{
+private:
+	void * const x;
+public:
+	explicit g_cast (void *y) : x(y) { }
+	g_cast (g_cast&& y) : x(y.x) { }
```

Comment:
This line can be changed to the mildly preferred `= delete` if two occurences of ternary operator in the tree gain a cast or lose the ternary operator.

Reply:
Let's fix the ternaries and put `= delete` here.